### PR TITLE
Supportupdate

### PR DESCRIPTION
**Pull request by dennisbrouwer91**
Added support for monitor_thresholds and monitoring_thresholds_windows needed for anomaly and process monitors

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -18,11 +18,12 @@ variable "include_tags" {
...
 variable "monitors" {
   type = map(object({
-    name       = string
-    message    = string
-    query      = string
-    thresholds = map(string)
-    type       = string
+    name                      = string
+    message                   = string
+    monitor_threshold_windows = map(string)
```
Comment (marwinbaumannsbp) [inline variables.tf]: I would suggest to make an object out of this:

```suggestion
    monitor_threshold_windows = object({
    trigger_window = string
    recovery_window    = string
  })
```

File: main.tf [hunk 2]
```diff
@@ -20,6 +21,13 @@ resource "datadog_monitor" "default" {
   type                = each.value.type
   tags                = local.tags
 
+  dynamic "monitor_threshold_windows" {
+    for_each = each.value.monitor_threshold_windows != null ? { create : true } : {}
+    content {
```
Comment (stefanwb) [inline main.tf:26]:
```suggestion
    for_each = each.value.monitor_threshold_windows != null ? { create : true } : {}
    
    content {
```
Could you add a newline here please?

File: main.tf [hunk 3]
```diff
@@ -32,4 +40,5 @@ resource "datadog_monitor" "default" {
       critical_recovery = try(each.value.thresholds["critical_recovery"], null)
     }
   }
+
```
Comment (stefanwb) [inline main.tf:43]:
```suggestion
```
Could you please remove this newline?

File: main.tf [hunk 4]
```diff
@@ -6,8 +6,9 @@ locals {
 resource "datadog_monitor" "default" {
   for_each = var.monitors
 
-  name                = each.value.name
-  message             = format(local.message, each.value.message)
+  name    = each.value.name
+  message = format(local.message, each.value.message)
+
```
Comment (stefanwb) [inline main.tf:11]:
```suggestion
```
Could you please remove this newline?

**Review by stefanwb — CHANGES_REQUESTED** [review]

**Review by stefanwb — CHANGES_REQUESTED** [review]
As final request, could you please rebase and squash your commits so that the history will look nicely?

File: variables.tf [hunk 5]
```diff
@@ -18,8 +18,12 @@ variable "include_tags" {
 
 variable "monitors" {
   type = map(object({
-    name       = string
-    message    = string
+    name    = string
+    message = string
+    monitor_threshold_windows = object({
```
Comment (svanharmelen) [inline variables.tf:23]: As this field is already in the `monitors` object, I think we can skip the `monitor` part here and just call this:
```suggestion
   threshold_windows = object({
```
It should also be reordered after the rename...

File: main.tf (hunk 3 above)
Comment (svanharmelen) [reply main.tf:43]: Guess this line should be between the two dynamic blocks as the are now glued together...

**Review by svanharmelen — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 5 above)
Comment (stefanwb) [reply variables.tf:23]: I agree, considering `thresholds` is also there we should keep it consistent.

**Pull request closed without merging**